sys/kern/ringbuf.c

```diff
@@ -28,6 +28,22 @@ static void consume(ringbuf_t *buf, unsigned bytes) {
     buf->tail = 0;
 }
 
+/* below two function are to help to track ringbuf pointers in case when it is
+ * changed by something else. e.q. DMA */
+void ringbuf_produce(ringbuf_t *buf, unsigned bytes) {
+  assert(buf->count + bytes <= buf->size);
```

Comment:
Maybe I am too paranoic.
```suggestion
  assert(buf->count  <= buf->size && bytes <= buf->size &&  buf->count  <= buf->size - bytes);
```